GUI.h

```diff
@@ -2,6 +2,15 @@
 
 #include "common.h"
 #include <filesystem>
+#include <functional>
+
+// GUI Element stuffs
+void handleMouseClick(int mouseX, int mouseY);
+void handleMouseMove(int mouseX, int mouseY);
+bool handleMouseWheel(int mouseX, int mouseY, int deltaY);
+void handleMouseRelease();
+typedef void (*OnClickCallback)(uint32_t);
```

Comment:
rewrite this as a `using` and using trailing return: `using OnClickCallback = auto (*)(uint32_t) -> void;`

`typedef` is bad enough, and using `typedef` for function pointers even more so; using trailing return makes it much more obvious what you're doing here